src/drivers/mediatek/mt8195/timer.c

```diff
@@ -44,7 +44,7 @@ int64_t platform_timer_set(struct timer *timer, uint64_t ticks)
 {
 	uint64_t time;
 	uint32_t hitimeout = ticks >> 32;
-	uint32_t lowtimeout = ticks & 0xFFFFFFFF;
+	uint32_t lowtimeout;
```

Comment:
I am having a hard time reconciling the two hi and low parts

````c
timer->hitimeout = ticks >> 32;
timer->lowtimeout = (ticks > time) ? ticks - time : 1;
````
Maybe it's just me but more comments wouldn't hurt

Reply:
This needs to be two patches then, one for the cppcheck fix and the other to to a low and high read (to read the high/low and ovoid the overflows). This is already done in other places in MTK timer and Intel timer.

Reply:
hmm I think that's ok for me.

```
timer->hitimeout = ticks >> 32;
timer->lowtimeout = (ticks > time) ? ticks - time : 1;
```


I will update this.